packages/plugin-dlx/sources/commands/dlx.ts

```diff
@@ -43,7 +74,35 @@ export default class DlxCommand extends BaseCommand {
 
       await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`package.json`)), `{}\n`);
       await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`yarn.lock`)), ``);
-      await xfs.writeFilePromise(ppath.join(tmpDir, toFilename(`.yarnrc.yml`)), `enableGlobalCache: true\n`);
+
+      const projectCwd = (await Configuration.findProjectCwd(this.context.cwd, Filename.lockfile))!;
+      const localConfigPath = ppath.join(projectCwd, toFilename(`.yarnrc.yml`));
+      const tmpDirConfigPath = ppath.join(tmpDir, toFilename(`.yarnrc.yml`));
+      if (await xfs.existsPromise(localConfigPath)) {
+        await xfs.copyFilePromise(localConfigPath, tmpDirConfigPath);
+        await Configuration.updateConfiguration(projectCwd, {
+          enableGlobalCache: false,
+        });
```

Comment:
You should update the configuration inside `tmpDir`, not inside `projectCwd` (which is the user's project). Note: I mistakenly said in my original comment `tmpDirConfigPath` instead of `tmpDir`.

Btw, why did you make it `enableGlobalCache: false` instead of `true`, like it originally was?

Reply:
Sorry, I feel like somewhere in implementing the workaround for removing config entries my brain got fried. Both points are completely right, I'll push the fix once I have @arcanis suggestion in.